### ISSUE TITLE
Windows 7/10: CPU 100%

### ISSUE THREAD

**Issue by szh7379** [monero-project/kovri#653]
As soon as the software is turned on, the computer is stuck and resources are soaring，I tried in windows7/10，both are stuck.

**Comment by anonimal** [comment]
>Windows

@szh7379 This could be any number of issues. I'm not ruling out kovri but you'll need to give more information.

- Did you build yourself or download the nightly build?
- Git revision / version?
- System details?
- Screenshot of resource manager window?

**Comment by szh7379** [comment]
Nightly version 20170601, 20170607， both are stuck.
System: windows 7/10, ram 4g,  i3,  with these it should happened
Every time i must force restart，i can't get a screenshot

**Comment by anonimal** [comment]
@szh7379 I cannot reproduce on Windows 10 Pro, x86-64, Xeon E5-1650 - nightly built against 5b1fd54306fb211ae0c923162d7b742d96e2d7c8, with over an hour of uptime. Screenshot attached. ![kovri-x86_64-win10](https://user-images.githubusercontent.com/12156341/26905186-69fc119c-4bd5-11e7-96f9-e618b3baab44.png)

As for the win32 nightly, I'll need more time to test. I also don't have a Win7 box readily available for testing until after Thursday.

@szh7379 What locale are you using?

**Comment by szh7379** [comment]
After click the install.bat, I running with a proxy to reseed on the desktop, nothing else.
I tried to copy the netDB from the i2p without proxy, than launch it, happened again.

**Comment by szh7379** [comment]
![default](https://user-images.githubusercontent.com/2356555/26911649-70089652-4c41-11e7-9eb0-9be9a135224f.png)
Run taskmgr first, I got a screenshot, it is  the memory, but cpu got a high consumption too.

**Comment by anonimal** [comment]
>After click the install.bat, I running with a proxy to reseed on the desktop, nothing else.
I tried to copy the netDB from the i2p without proxy, than launch it, happened again.

I'm sorry, could you rephrase this? I don't quite understand.  Also, do you happen to have a build environment setup (MSYS2/MinGW-64) available for debugging?

**Comment by anonimal** [comment]
> Your reported resources

Two important notes:

1. After testing on a win32 VM (Win 10, same hardware), I still **cannot** reproduce this issue.
2. Memory consumption is normal / within expected parameters on both systems. Screenshot attached.

![kovri-x86_64-win10 memory](https://user-images.githubusercontent.com/12156341/26914498-2b2fe86c-4c11-11e7-9912-580f3886823c.png)

**Comment by anonimal** [comment]
@szh7379 could you please open powershell or cmd, run kovri from the command line, and take a screenshot?

**Comment by szh7379** [comment]
In win 10, it is totally stuck, the same hardware. i can't get anything, i can take a photo, but the comsuption can't see. because it is not what happened right now.
![img20170608162906](https://user-images.githubusercontent.com/2356555/26919271-b9ecf0b6-4bfa-11e7-90d3-323a990c088f.jpg)
when I run kovri first time, it told me need some lib，so i download SSLeay.dll ssleay32.dll, should these be the reason?

and sorry i get no build environment for debug.

**Comment by szh7379** [comment]
kill every app. get the screenshots
memory:
![memory](https://user-images.githubusercontent.com/2356555/26938441-d3e6c53a-4ca6-11e7-841f-546d9d76c73e.jpg)
overview:
![overview](https://user-images.githubusercontent.com/2356555/26938473-e491c5a6-4ca6-11e7-83d3-f583d25c3e92.jpg)

**Comment by anonimal** [comment]
@szh7379 thank you very much for sending the photo screenshot.

>so i download libeay32.dll SSLeay.dll ssleay32.dll, should these be the reason?

1. Judging by the screenshot, you were able to successfully reseed (which uses those OpenSSL libraries)  so, I'm unsure how they are related at this point.
2. You shouldn't have needed to download those libs because we ship them statically with the binary - or at least we used to until something happened with CMake or GCC and/or our setup of which I've just now noticed (in other words, the Kovri binary is now looking for OpenSSL dynamically on Windows).

I will need to spend time looking into point 2. As for point 1, can you do the following:

1. Open cmd or powershell
2. Run `kovri.exe --log-level 5`
3. Upload the latest log file. The logs will be in your app data directory. To find that, open cmd and `cd %AppData%\Kovri\logs`

**Comment by szh7379** [comment]
[log.txt](https://github.com/monero-project/kovri/files/1062056/log.txt)
this it the log file

**Comment by anonimal** [comment]
1. The log shows a runtime of 9 minutes. Does your application stall after 9 minutes or immediately?
2. This log shows that you're loading routers from disk. Can you please do a fresh install (just run the installer again), run `kovri.exe --log-level 5`, and upload the new log? Thank you.

**Comment by szh7379** [comment]
after fresh install, it don't stall immediately, but when the reseed be fectched successfuly, it happen again.

**Comment by szh7379** [comment]
[x.txt](https://github.com/monero-project/kovri/files/1062937/x.txt)
This is the log file of the fresh install. It ended in the time the when kovri stalled.  i have to force close my system.  i don't know is it enough for the debugging.

**Comment by szh7379** [comment]
[log.txt](https://github.com/monero-project/kovri/files/1064776/log.txt)
This is a unbroken log file. I think the reason is reseeding. 
kovri stall at 2017.06.09 16:56:04.468483, and lasted 8 minutes. than i close the cmd, kovri was reseeding all that time.

**Comment by anonimal** [comment]
What doesn't make sense is, in your [first log](https://github.com/monero-project/kovri/files/1062056/log.txt), you were obviously able to reseed and populate the network database. In the last two logs [here](https://github.com/monero-project/kovri/files/1062937/x.txt) and [here](https://github.com/monero-project/kovri/files/1064776/log.txt), it appears that a thread is **possibly** dead/live-locked with another, with one or more causing 100% CPU utilization (I won't know for certain until I have time to look). A quick glance shows that one of the threads attempts to validate the SU3 signer ID,  using `std::regex` no less. One of the reasons I asked about what locale you were using here https://github.com/monero-project/kovri/issues/653#issuecomment-306951498 was because regex has proven to be problematic with certain locales (as seen here 0ce92bdb1563db22819596206314a81989e37c67).

For all of the aforementioned issues, I would need to set time aside to focus on them as I've only grazed this issue while doing a lot of other project work this week. What would be helpful is if you can setup a developer environment to help debug / test patches. Otherwise this issue may go by the way-side until I can reproduce the issue on my available machines (of which I have yet to be able to reproduce) or if another developer gets involved. Dev setup instructions are [here](https://github.com/anonimal/kovri-docs/blob/b3572e7eacd1c2d989d180a187d9a050875d0c2d/end-user/BUILDING.md#windows-msys2mingw-64) and if you need help setting up, please let me know.

What would also help is if you could some how monitor thread usage using whatever Windows tools are available. I don't have any immediate recommendations but I'm sure they're out there.

Sidenote: have you also tested on Linux?

**Comment by szh7379** [comment]
Why can i reseed in first log is i copy the netDB directory from the i2p into kovri, the second and the last one is the fresh install and running with a VPN, because i2p can't connect the KAD network in the first running in my place. the locale is zh-CN. 
Sorry, i am not a programmer, have no Dev environment, get no Linux system. that is all what i can do
It is ok, focus on these problems what you can test in you environment.

**Comment by szh7379** [comment]
And one more thing, if i copy the netDB Dir to the kovri appdata Dir,  stalling will happen even i shut down the network.
I think we can close the isuue.😀
